exopy_hqc_legacy/instruments/drivers/visa/tektro_awg.py

```diff
@@ -713,7 +755,8 @@ def running(self, value):
         self.clear_output_buffer()
         if value in ('RUN', 1, 'True'):
             self.write('AWGC:RUN:IMM')
-            if self.ask_for_values('AWGC:RST?')[0] not in (1, 2):
+            values = self.query('AWGC:RST?', format=2, delay=self.delay)
```

Comment:
query gives you a string not a list of values hence the branch will never be entered.